dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java

```diff
@@ -441,69 +450,82 @@ private void selectMultipleItems(final List<String> rowNameList, final TreeTable
      * @param localPath path where selected files need to transfer
      */
     private void startGetJob(final List<Ds3TreeTableValueCustom> listFiles,
-            final Path localPath) {
-        final Ds3GetJob getJob = ds3GetJobFactory.createDs3GetJob(listFiles, localPath);
-        getJob.setOnSucceeded(SafeHandler.logHandle(event -> {
-            LOG.info("Get Job completed successfully");
-            refreshFileTreeView();
-        }));
-        getJob.setOnFailed(SafeHandler.logHandle(event -> {
-            final Throwable exception = event.getSource().getException();
-            LOG.error("Get Job failed", exception);
-            loggingService.logMessage("Get Job failed with message: " + exception.getMessage(), LogType.ERROR);
-            refreshFileTreeView();
-        }));
-        getJob.setOnCancelled(SafeHandler.logHandle(cancelEvent -> {
-            //Cancellation of a job started
-            final Ds3CancelSingleJobTask ds3CancelSingleJobTask = new Ds3CancelSingleJobTask(getJob.getJobId().toString(), endpointInfo, jobInterruptionStore, JobRequestType.GET.toString(), loggingService);
-            ds3CancelSingleJobTask.setOnFailed(SafeHandler.logHandle(event -> LOG.error("Failed to cancel job")));
-            ds3CancelSingleJobTask.setOnSucceeded(SafeHandler.logHandle(event -> {
-                LOG.info("Get Job cancelled");
-                loggingService.logMessage("GET Job Cancelled", LogType.INFO);
-                refreshFileTreeView();
-            }));
-            workers.execute(ds3CancelSingleJobTask);
-        }));
-        jobWorkers.execute(getJob);
+            final Path localPath, final Session session) {
+        listFiles.stream()
+                .map(Ds3TreeTableValueCustom::getBucketName)
+                .distinct()
+                .forEach(bucket -> {
+                    final ImmutableList<kotlin.Pair<String, String>> fileAndParent = listFiles.stream()
+                            .filter(ds3TreeTableValueCustom -> Objects.equals(ds3TreeTableValueCustom.getBucketName(), bucket))
+                            .map(ds3 -> new kotlin.Pair<>(
+                                    ds3.getFullName(),
+                                    ds3.getParent()))
+                            .collect(GuavaCollectors.immutableList());
+                    final JobTaskElement jobTaskElement = new JobTaskElement(settingsStore, loggingService, dateTimeUtils, session.getClient(), jobInterruptionStore, savedJobPrioritiesStore, resourceBundle);
```

Comment:
Similar comment here about eliminating the need to pass in session when we only use the client from it.